wire/msgcftypes.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2017 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package wire
+
+import "io"
+
+// FilterType is used to represent a filter type.
+type FilterType uint8
+
+const (
+	// GCSFilterRegular is the regular filter type.
+	GCSFilterRegular FilterType = iota
+
+	// GCSFilterExtended is the extended filter type.
+	GCSFilterExtended
+)
+
+// MsgCFTypes is the cftypes message.
+type MsgCFTypes struct {
+	SupportedFilters []FilterType
+}
+
+// BtcDecode decodes r using the wire protocol encoding into the receiver.
+// This is part of the Message interface implementation.
+func (msg *MsgCFTypes) BtcDecode(r io.Reader, pver uint32) error {
+	// Read the number of filter types supported.
+	count, err := ReadVarInt(r, pver)
+	if err != nil {
+		return err
+	}
+
+	// Read each filter type.
+	msg.SupportedFilters = make([]FilterType, count)
+	for i := uint64(0); i < count; i++ {
+		var filterType uint8
+		err = readElement(r, &filterType)
```

Comment:
Same as above regarding the cast or addition of `*FilterType` to `readElement`.

Also, to cut down on the number of allocations, please read it directly into the array element.